### PR TITLE
fix: `time_live` should be set at 3am in local time for all surfaceGuid

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal
`time_live` should be set to 3am in the local timezone
- en-US: EST 3am
- de-de: Berline time 3am
- en-intl: 3am IST
- en-gb: 3am GMT

## Implementation Decisions
- `new Date()` sets date in machine time when we try to manually set hours. so used `Date.UTC` 
- added spec test to make sure we are able to set time in 3am (in local timezone)
- timezone doc: https://getpocket.atlassian.net/wiki/spaces/PE/pages/2584150049/Pocket+Shared+Data#Source-of-Truth.5
- validated in integration test as well

## Feedback on:
- manually setting `hours` is not an accurate solution with day light savings (e.g `gb` changes between time/GMT time for summer-winter months). my understanding is we want the new tab to get changed around 3am in local timezone, but in case of daylight saving - this would still constantly change every 24 hrs (but on 2am instead). 
- we mainly focus on U.S, where we want the new tab content to change at 12 am PST, i.e 3am EST 

Ticket: https://getpocket.atlassian.net/browse/INFRA-510

### CODE REVIEW

File: curation-migration-datasync/helpers/timeConverter.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { ScheduledSurfaceGuid } from '../dynamodb/types';
+
+/**
+ * The given scheduled date is converted to 3am in their local time
+ * for the corresponding NewTab, and then converted to unix epoc timestamp.
```
Comment (nina-py) [inline curation-migration-datasync/helpers/timeConverter.ts:5]: Nitpick: typo in `epoch`.

File: curation-migration-datasync/helpers/timeConverter.spec.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { getLocalTimeFromScheduledDate } from './timeConverter';
+import { ScheduledSurfaceGuid } from '../dynamodb/types';
+
+describe('time converter', function () {
+  //for timestamp conversion: https://www.epochconverter.com/
+  const testDate = '2022-06-29';
+  const estTimestamp = 1656486000; //add 7 hrs to the given date
+  const berlinTimestamp = 1656464400; //add one hour to the given date
+  const istTimestamp = 1656451800; //subtract one day and set to 930 pm
+  const gmtTimestamp = 1656471600; //add 3 hours with given date
```
Comment (nina-py) [inline curation-migration-datasync/helpers/timeConverter.spec.ts:10]: Nitpick: all three comments that start with "add X hours" should use "to the given date".

File: curation-migration-datasync/helpers/timeConverter.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * For e.g scheduledDate `2022-06-01` is converted to:
+ * istTimestamp = '2022-05-31 00:21:30'; // subtract one day and set to 930 pm
+ * berlinTime = '2022-06-01 00:01:00'; //add one hour
+ * ukTime = '2022-06-01 00:03:00'; // add 3 hours with date
+ * estTime = '2022-06-01 00:07:00'; //add 7 hrs with date
```
Comment (nina-py) [inline curation-migration-datasync/helpers/timeConverter.ts:11]: Nitpick: "with date" should be replaced with "to the timestamp provided" or similar.

**Review by nina-py — APPROVED** [review]

File: curation-migration-datasync/helpers/timeConverter.ts (hunk 3 above)
Comment (sri-kirsh) [reply curation-migration-datasync/helpers/timeConverter.ts]: the function returns the `timestamp` and gets the date as input..so we are adding/subtracting hrs to the Date object and converting it to epoch timestamp, updated the function doc to make it less confusing

File: curation-migration-datasync/helpers/timeConverter.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ * Function to generate Date object for given scheduledDate, hours, mins, secs
+ * Date.UTC is required, otherwise date object takes machine time while manually setting hours
+ * month offset starts from 0
+ * @param scheduledDate
+ * @param hours
+ * @param mins
+ * @param secs
+ */
+function getDate(scheduledDate: string, hours = 0, mins = 0, secs = 0): Date {
+  const d = scheduledDate.split('-');
```
Comment (Herraj) [inline curation-migration-datasync/helpers/timeConverter.ts:58]: minor nitpick: can we change the `d` varialbe to be more descriptive or I guess add a comment

**Review by Herraj — APPROVED** [review]

**Pull request merged**